pandas/core/indexes/range.py

```diff
@@ -897,12 +910,19 @@ def symmetric_difference(
             result = result.rename(result_name)
         return result
 
+    def _join_empty(
+        self, other: Index, how: JoinHow, sort: bool
+    ) -> tuple[Index, npt.NDArray[np.intp] | None, npt.NDArray[np.intp] | None]:
+        if other.dtype.kind == "i":
```

Comment:
Why is this specific to just this dtype? I am somewhat wary of doing this one-off in a function like this

Reply:
We can only convert `Index[int]` to `RangeIndex` (which what `_shallow_copy` will do if possible) to use later in `_join_empty` to ensure that returns a `RangeIndex`